### DISCUSSION TITLE
[Question]: Support Coral.ai Accelerator ?

### DISCUSSION THREAD

**Discussion by Nonobis** [immich-app/immich#6872] [Feature Request]
### Feature detail

I don't known if it's  a good idea or not ... i read a few article on this hardware and this seem really cool for running tensorflow locally.

Maybe support this type of hardware (https://coral.ai/products/accelerator) could help your AI model ? support for a specifc hardware with relative low cost / power usage for accelerate tensorflow seem to be a good idea.

Maybe for a better object/facial recognition ?

### Platform

Server

**Comment by alextran1502** [comment]
This is very specific and at the moment and we don't have the capacity for this yet. I will keep this in mind, and if there are more requests for this, then I will revisit this issue.

**Comment by jwynn6** [comment]
+1 Would love this. I have older hardware that doesn't support the AVX and AVX2 commands, and a couple of coral accelerators for use with the Frigate project. I would love to redirect one of them to be used here.

**Comment by DrMxrcy** [comment]
+1 This would be amazing to have, especially on a little Pi

**Comment by arcoast** [comment]
I'd also like to see this, with energy prices climbing, anything to offset CPU use would be a bonus.

**Comment by Hukuma1** [comment]
Would be amazing! I bought one for Frigate, but since it's dual Edge TPU, would love the use the second for Immich. +1

**Comment by jameswynn** [comment]
I'm also very keen on this idea! There are dozens of selfhosters that already have these for Frigate. Dozens!

**Comment by mike-lloyd03** [comment]
Literally dozens of us!

**Comment by hvddrift** [comment]
> +1 Would love this. I have older hardware that doesn't support the AVX and AVX2 commands, and a couple of coral accelerators for use with the Frigate project. I would love to redirect one of them to be used here.

I am in exactly the same boat as jwynn6. CPU has no AVX but I do have a coral accelerator. Love to see this implemented if you have the time.

**Comment by jagjordi** [comment]
I also thing this would be an extremely nice feature. In the upcoming years we will probably see more AI edge accelerators on the market so it would be good to ride those waves

**Comment by franciscofsales** [comment]
+1, this would be great

**Comment by adambeck7** [comment]
+1. There's definitely a lot of overlap between the people who self-host and the people who own Corals now. Largely because of Frigate for many of us.

**Comment by joy4eg** [comment]
With release v1.56.0 we have now `Facial Recognition`. Probably it's a good reason/opportunity to re-think about that issue :) Thank you for your amazing work.

**Comment by ghost** [comment]
No avx on my older hardware either and with more functions such as facial recognition being added that require avx support for the coral would be awesome.

**Comment by 4d1in3** [comment]
+1 running on older hardware, coral support would be awesome

**Comment by Nonobis** [comment]
Got a Google Coral TPU, if needed i can make a few test :)

**Comment by Schluggi** [comment]
Any progress here?

**Comment by tehniemer** [comment]
One of the dozens here loving this idea!

**Comment by akropp** [comment]
Make that dozens and one!

**Comment by erikhubers** [comment]
Sorry for asking but, how does the PR mentioned above close / resolve this issue?

**Comment by mrnoisytiger** [comment]
+1 for supporting a Coral TPU. I'm running Immich on a older Pentium and while it's been good for general use, I had to turn off ML to avoid melting the CPU back into raw silicon.

**Comment by AlessandroTischer** [comment]
For what it matters, I would love to see coral rupport on Immich!

**Comment by lukasMega** [comment]
+1 for Coral support & reopening this issue (:

**Comment by Surgikill** [comment]
Same here, +1 for coral support.

**Comment by bo0tzz** [comment]
It's currently not feasible for us to support this, see https://github.com/immich-app/immich/discussions/73#discussioncomment-6891382

**Reply by psychowood** [reply]
For complete reference to the dozens (+1, including me) of TPU users

https://github.com/microsoft/onnxruntime/issues/10248

**Comment by noelhibbard** [comment]
The Coral is dirt cheap for what it does. I have a USB Coral and a Dual Edge.... because of Frigate of course. So, add me to the dozen that owns a few.

**Comment by Joentje** [comment]
Same here, +1 for coral support.

**Comment by brdweb** [comment]
+1 here

**Comment by crypto-smoke** [comment]
Another user that would love coral tpu support. Running immich on a pi is great but slapping a GPU in there for the ML tasks isn't feasible. Corals are cheap and power efficient. I hope it's still feasible to add this as it would really help set immich even further apart from the competition.

**Comment by brdweb** [comment]
Yet another +1 here. Same thing. Bought for Frigate but would love to leverage it to reduce usage on CPU/GPU for immich.

**Comment by cdoublejj** [comment]
i'd offer to buy the Devs Coral cards but, it looks like Louisrossman or FUTO can now be the generous donor. look forwards to seeing HW acceleration for AI CPU offloading. Congratulation guys!

**Comment by Dud3z** [comment]
Same here. i have a dual tpu.. one half is used for frigate rn

**Comment by ruswerner** [comment]
I'm also using a coral USB with frigate in docker and it works great (perfect) on low-end hardware. +1 on supporting it in this project.

**Comment by nedich** [comment]
+1 for coral support!! 
already got the coral, would be great in immich not to fry the cpu but to use the coral instead.

**Comment by bo0tzz** [comment]
As mentioned in https://github.com/immich-app/immich/discussions/73#discussioncomment-9414732, coral support is just not possible and so will never happen.
